iuriMorari/src/main/java/AdminUser.java

```diff
@@ -0,0 +1,16 @@
+import java.util.Map;
+
+public class AdminUser extends AbstractUser {
+
+    public AdminUser(String username, Map<String, Address> addresses) {
+        super(username, addresses);
+    }
+
+    public void approveOrder(AbstractOrder abstractOrder) {
+        System.out.println("Order " + abstractOrder.generatedID() + " is approved");
```

Comment:
id should be of the order to be approved/rejected, why do you generate it here?

Reply:
As I understood, now its "getId()"